pkg/accounting/tally/tally_test.go

```diff
@@ -59,6 +59,58 @@ func TestDeleteRawBefore(t *testing.T) {
 	}
 }
 
+func TestOnlyInline(t *testing.T) {
+	testplanet.Run(t, testplanet.Config{
+		SatelliteCount: 1, StorageNodeCount: 6, UplinkCount: 1,
+	}, func(t *testing.T, ctx *testcontext.Context, planet *testplanet.Planet) {
+		tallySvc := planet.Satellites[0].Accounting.Tally
+		uplink := planet.Uplinks[0]
+
+		// Setup: create data for the uplink to upload
+		expectedData := make([]byte, 1*memory.KiB)
+		_, err := rand.Read(expectedData)
+		require.NoError(t, err)
+
+		// Setup: get the expected size of the data that will be stored in pointer
+		uplinkConfig := uplink.GetConfig(planet.Satellites[0])
+		expectedTotalBytes, err := encryption.CalcEncryptedSize(int64(len(expectedData)), uplinkConfig.GetEncryptionScheme())
+		require.NoError(t, err)
+
+		// Setup: The data in this tally should match the pointer that the uplink.upload created
+		expectedTally := accounting.BucketTally{
+			Segments:       1,
+			InlineSegments: 1,
+			Files:          1,
+			InlineFiles:    1,
+			Bytes:          expectedTotalBytes,
+			InlineBytes:    expectedTotalBytes,
+			MetadataSize:   111, // brittle, this is hardcoded since its too difficult to get this value progamatically
+		}
+
+		// Execute test: upload a file, then calculate at rest data
+		expectedBucketName := "testbucket"
+		err = uplink.Upload(ctx, planet.Satellites[0], expectedBucketName, "test/path", expectedData)
+		assert.NoError(t, err)
+
+		// Run calculate twice to test unique constraint issue
+		for i := 0; i < 2; i++ {
+			latestTally, actualNodeData, actualBucketData, err := tallySvc.CalculateAtRestData(ctx)
+			require.NoError(t, err)
+			assert.Len(t, actualNodeData, 0)
+
+			_, err = planet.Satellites[0].DB.Accounting().SaveBucketTallies(ctx, latestTally, actualBucketData)
+			require.NoError(t, err)
```

Comment:
Unless we want to check if SaveBucketTallies worked as expected, we don't need to call it. In TestCalculateAtRestData, we just check the result of CalculateAtRestData, not the result of the db methods.

Reply:
The error that this PR fixes was coming from SaveBucketTallies

Reply:
unique constraint on attempting to insert into the database

Reply:
gotcha